src/utils/enableWhen.ts

```diff
@@ -0,0 +1,142 @@
+import _ from 'lodash';
+import type {
+    QuestionnaireItemEnableWhenAnswer,
+    QuestionnaireItemAnswerOption,
+    QuestionnaireItemEnableWhen,
+} from 'shared/src/contrib/aidbox';
+import * as yup from 'yup';
+
+type EnableWhenAnswerTypes = 'Coding' | 'string' | 'integer' | 'boolean';
+export type EnableWhenOperator = 'exists' | '=' | '!=' | '>' | '<' | '>=' | '<=';
+export type EnableWhenValueType = string | number | boolean;
+type EnableWhenAnswerTypesMap = Record<EnableWhenAnswerTypes, { path: string }>;
+
+const VALUES_TYPES: (keyof EnableWhenAnswerTypesMap)[] = ['Coding', 'string', 'integer', 'boolean'];
+const VALUES_TYPES_PATH_MAP: EnableWhenAnswerTypesMap = {
+    Coding: { path: 'Coding' },
+    string: { path: 'string' },
+    integer: { path: 'integer' },
+    boolean: { path: 'boolean' },
+};
+
+const ENABLE_WHEN_OPERATORS_MAP: Record<
+    EnableWhenOperator,
+    (a: EnableWhenValueType, b: EnableWhenValueType) => boolean
+> = {
+    exists: (a, b) => !!a === !!b,
+    '=': (a, b) => _.isEqual(a, b),
```

Comment:
better to use [isValueEqual](https://github.com/beda-software/sdc-ide/blob/master/sdc-qrf/src/utils.ts#L412) instead of isEqual, but it works with {Coding: {}} structures. The main idea of this function, that coding should be compared by code & system and **not display**.